## Import dataset

In [9]:
import pandas as pd

df = pd.read_csv("../data/twcs_chat_logs.csv")
df.head()


,tweet_id,author_id,inbound,created_at,text,response_tweet_id,in_response_to_tweet_id
0,1,sprintcare,False,Tue Oct 31 22:10:47 +0000 2017,@115712 I understand. I would like to assist y...,2,3.0
1,2,115712,True,Tue Oct 31 22:11:45 +0000 2017,@sprintcare and how do you propose we do that,NaN,1.0
2,3,115712,True,Tue Oct 31 22:08:27 +0000 2017,@sprintcare I have sent several private messag...,1,4.0
3,4,sprintcare,False,Tue Oct 31 21:54:49 +0000 2017,@115712 Please send us a Private Message so th...,3,5.0
4,5,115712,True,Tue Oct 31 21:49:35 +0000 2017,@sprintcare I did.,4,6.0


## Inspect dataset 
1. check basic info
2. checl for missing values
3. Few Randowm rows

In [2]:
# 1. basic info
df.info


<bound method DataFrame.info of          tweet_id   author_id  inbound                      created_at  \
0               1  sprintcare    False  Tue Oct 31 22:10:47 +0000 2017   
1               2      115712     True  Tue Oct 31 22:11:45 +0000 2017   
2               3      115712     True  Tue Oct 31 22:08:27 +0000 2017   
3               4  sprintcare    False  Tue Oct 31 21:54:49 +0000 2017   
4               5      115712     True  Tue Oct 31 21:49:35 +0000 2017   
...           ...         ...      ...                             ...   
2811769   2987947  sprintcare    False  Wed Nov 22 08:43:51 +0000 2017   
2811770   2987948      823869     True  Wed Nov 22 08:35:16 +0000 2017   
2811771   2812240      121673     True  Thu Nov 23 04:13:07 +0000 2017   
2811772   2987949      AldiUK    False  Wed Nov 22 08:31:24 +0000 2017   
2811773   2987950      823870     True  Tue Nov 21 22:01:04 +0000 2017   

                                                      text response_tweet_id  \

In [4]:
# total missing values
df.isna().sum()


tweet_id                         0
author_id                        0
inbound                          0
created_at                       0
text                             0
response_tweet_id          1040629
in_response_to_tweet_id     794335
dtype: int64

In [5]:
# soem random rows
df.sample(5)

,tweet_id,author_id,inbound,created_at,text,response_tweet_id,in_response_to_tweet_id
1717566,1871796,115913,True,Fri Oct 20 02:03:40 +0000 2017,@115725 @369031 Oh wait... yep... he did. #can...,1871795,1871797.0
275839,316084,ChipotleTweets,False,Fri Oct 20 22:56:34 +0000 2017,"@191469 You're welcome, Emily. -Gabe",NaN,316083.0
44933,55919,128684,True,Thu Nov 23 00:47:14 +0000 2017,Like my phone was just on a smooth 60% go on s...,55918,NaN
394667,450089,221685,True,Thu Nov 30 20:29:45 +0000 2017,@Tesco Girl cleaning bakery was great. Think s...,450088,450090.0
1472240,1617452,495970,True,Sun Nov 05 14:08:54 +0000 2017,@sainsburys I’ve purchased Jam Donuts from the...,1617450,NaN


## Understanding the meaning of each column


tweet_id- Unique ID for each tweet  
author_id- Who wrote the tweet (customer or company)  
inbound- True = customer message, False = company reply  
text- The actual tweet content  
created_at- Timestamp of the tweet   
response_tweet_id-  the tweet(s) that responded to it (could be multiple id's respondign to the tweet)  
in_reply_to_status_id- Which tweet this message is replying to (helps build conversation threads)  

In [6]:
# totla no. of rows
len(df)


2811774

## Create a tiny working sample 
To keep things fast while learning, let's create a smaple of this csv file

In [7]:
df_small = df.sample(5000, random_state=42)
df_small.to_csv("../data/twcs_small.csv", index=False)


## Quick mental model (super helpful)
#### tweets like nodes in a tree:

```
tweet A
 ├── tweet B (reply)
 ├── tweet C (reply)
 │     └── tweet D (reply to C)
```
- response_tweet_id tells you children
- in_response_to_tweet_id tells you parent  
This is how we’ll build conversation IDs later.

## Missing values (your isna output)
We saw:  
response_tweet_id → ~1,040,629 missing

in_response_to_tweet_id → ~794,335 missing

This is normal. Beacause:
- Many tweets are root tweets (not replies)
- Many tweets have no replies
- Some conversations are one-way  
This is expected in customer support.